### PR TITLE
fix: address axelar transceiver audit comments

### PR DESCRIPTION
**Pull request by Foivos**
Address Axelar transceiver audit comments

### CODE REVIEW

File: src/axelar/AxelarTransceiver.sol [hunk 1]
```diff
@@ -79,11 +79,20 @@ contract AxelarTransceiver is IAxelarTransceiver, AxelarGMPExecutable, Transceiv
...
         string calldata transceiverAddress
     ) external virtual onlyOwner {
+        if (chainId == 0 || bytes(chainName).length == 0 || bytes(transceiverAddress).length == 0) {
+            revert InvalidChainIdParams();
+        }
+
         AxelarTransceiverStorage storage slot = _storage();
+        if (bytes(slot.idToAxelarChainId[chainId]).length != 0) revert ChainIdAlreadySet(chainId);
+
+        if (slot.axelarChainIdToId[chainName] != 0) revert AxelarChainIdAlreadySet(chainName);
```
Comment (milapsheth) [inline src/axelar/AxelarTransceiver.sol:89]:
```suggestion

        if (bytes(slot.idToAxelarChainId[chainId]).length != 0) revert ChainIdAlreadySet(chainId);

        if (slot.axelarChainIdToId[chainName] != 0) revert AxelarChainIdAlreadySet(chainName);

```

**Review by milapsheth — DISMISSED** [review]

**Review by pleasew8t — DISMISSED** [review]

File: src/axelar/AxelarTransceiver.sol [hunk 2]
```diff
@@ -185,7 +185,10 @@ contract AxelarTransceiver is IAxelarTransceiver, AxelarGMPExecutable, Transceiv
     ) internal virtual override {
         AxelarTransceiverStorage storage slot = _storage();
         uint16 sourceChainId = slot.axelarChainIdToId[sourceChain];
-        if (sourceChainId == 0 || slot.transceiverAddressToId[sourceAddress] != sourceChainId) {
+        if (
+            sourceChainId == 0
+                || slot.idToTransceiverAddressHash[sourceChainId] != keccak256(bytes(sourceAddress))
```
Comment (djb15) [inline src/axelar/AxelarTransceiver.sol:190]: Why do we need to use a hash for values of this mapping? Message delivery will be pretty expensive if we need to compute a hash every time

Comment (milapsheth) [reply src/axelar/AxelarTransceiver.sol:190]: The transceiver address hex string is more than 32 bytes long. So loading it as a string value requires loading multiple storage slots, which is more expensive than hashing the string

**Review by milapsheth — APPROVED** [review]

File: src/axelar/AxelarTransceiver.sol (hunk 2 above)
Comment (djb15) [reply src/axelar/AxelarTransceiver.sol:190]: Ah yes fair point!

**Review by djb15 — APPROVED** [review]

File: src/axelar/AxelarTransceiver.sol (hunk 2 above)
Comment (Foivos) [reply src/axelar/AxelarTransceiver.sol:190]: it is cheaper than looking up the entire address as a string, so this is the cheapest safe way to do it. (mappings of strings compute the hash of them every time too)

**Pull request merged**